builder/js/ncf.js

```diff
@@ -798,7 +798,9 @@ $scope.groupMethodsByCategory = function () {
       param=""
     }
     // do not canonify what is between ${ }
-    param = param.replace(/\\'/g, "'").replace(/\\"/g, '"').replace(/[^\${}\w](?![^{}]+})|\$(?!{)/g,"_");
+    // regex description: replace every valid sequence followed by and invalid char with the sequence followed by _
+    //                    a valid sequence is either a word or a ${} expression
+    param = param.replace(/\\'/g, "'").replace(/\\"/g, '"').replace(/(\w+|\$\{\w+?\})?[^\$\w]/g,"$1_");
```

Comment:
I think this will not work with inner variables (${value{other_value}}) or array access (${array[index]})